From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martijn Muijsers <martijnmuijsers@live.nl>
Date: Wed, 10 Jan 2024 20:54:42 +0100
Subject: [PATCH] TODO Backwards plugin compatibility mode - Add matchable
 Material values

Adds additional values can be matched on in `Material.matchMaterial`:
the namespaced key and the namespaced key path of the `Material` if the backwards plugin compatibility mode were reversed.

License: GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0.html)
Fiddle - https://fiddlemc.org

diff --git a/src/main/java/org/fiddlemc/fiddle/material/MaterialNameAndKeyConversion.java b/src/main/java/org/fiddlemc/fiddle/material/MaterialNameAndKeyConversion.java
index 582b539d165454611c5bb74a7bae78d5eaf6fae3..fe14459638ae832f98594a95057a3b69246c661a 100644
--- a/src/main/java/org/fiddlemc/fiddle/material/MaterialNameAndKeyConversion.java
+++ b/src/main/java/org/fiddlemc/fiddle/material/MaterialNameAndKeyConversion.java
@@ -76,6 +76,9 @@ public final class MaterialNameAndKeyConversion {
             case 0 -> material.isLegacy() ? null : material.getKey().toString();
             case 1 -> material.isLegacy() ? null : material.getKey().getKey();
             case 2 -> material.name();
+            // Fiddle start - backwards plugin compatibility mode - add matchable Material values
+            // TODO add a case 3 and 4: the namespaced key and namespaced key path of this Material if the backwards plugin compatibility mode were reversed
+            // Fiddle end - backwards plugin compatibility mode - add matchable Material values
             default -> throw new IllegalArgumentException("Invalid valueIndex: " + valueIndex);
         };
     }
